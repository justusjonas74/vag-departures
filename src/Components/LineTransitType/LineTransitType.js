import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './LineTransitType.css';

/* ----------------- */
/*    EXPORTS        */
/* ----------------- */

class LineTransitType extends Component {
  styleTransitType(departureTransitType) {
    const switchTerm = departureTransitType ? departureTransitType.toLowerCase() : ""
    switch (switchTerm) {
      case 'ubahn':
        return {
          cssClass: 'ubahn',
          symbolStr: 'fa-train'
        };
      case 'tram':
        return {
          cssClass: 'tram',
          symbolStr: 'fa-subway'
        };
      case 'bus':
        return {
          cssClass: 'bus',
          symbolStr: 'fa-bus'
        };
      default:
        return {
          cssClass: 'bus',
          symbolStr: 'fa-bus'
        };
    }
  }

  render() {
    const departureLine = this.props.departureLine
    const departureTransitType = this.props.departureTransitType;
    const typeStyles = this.styleTransitType(departureTransitType);

    return (
        <span className="departureTransitType">
          <i className={'fa ' + typeStyles.symbolStr} aria-hidden="true"></i>
          <span className={'departureLine '+ typeStyles.cssClass}>{departureLine}</span>
        </span>
    )
  }
}

LineTransitType.propTypes = {
  departureTransitType: PropTypes.string,
  departureLine: PropTypes.string
}

export default LineTransitType