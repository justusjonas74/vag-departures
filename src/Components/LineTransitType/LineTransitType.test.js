import React from 'react';
import { shallow, mount } from 'enzyme';
import LineTransitType from './LineTransitType';

describe('LineTransitType', () => {
  let props;
  let mountedLineTransitType;
  const lineTransitType = () => {
    if (!mountedLineTransitType) {
      mountedLineTransitType = mount(
        <LineTransitType {...props} />
      );
    }
    return mountedLineTransitType;
  }

  beforeEach(() => {
    props = {
      departureLine: undefined,
      departureTransitType: undefined
    };
    mountedLineTransitType = undefined;
  });

  // UNIT TESTS
  it('should render correctly  without props', () => {
    shallow(<LineTransitType />);
  });
  it("always renders a span", () => {
    const spans = lineTransitType().find("span");
    expect(spans.length).toBeGreaterThanOrEqual(1);
  });
  describe("the rendered span", () => {
    it("contains everything else that gets rendered", () => {
      const spans = lineTransitType().find("span");
      const wrappingSpan = spans.first();
      expect(wrappingSpan).toEqual(lineTransitType().children());
    });
  });
  describe("when props are defined", () =>{
    describe("should render bus correctly", ()=>{
      beforeEach(()=>{
        props.departureLine = "68"
        props.departureTransitType = "Bus"
      })
      it('should render the correct line name', ()=>{
        const mountedLineTransitType = lineTransitType()
        const span = mountedLineTransitType.find('.departureLine')
        // console.log('SPAN: ' + span.text())
        expect(span.text()).toEqual('68')
      })
      it('should render a correct symbol', ()=>{
        const mountedLineTransitType = lineTransitType()
        const iElement = mountedLineTransitType.find("i.fa")
        expect(iElement.hasClass("fa-bus")).toEqual(true)
      })
      
    })
    describe("should render tram correctly", ()=>{
      beforeEach(()=>{
        props.departureLine = "68"
        props.departureTransitType = "Tram"
      })
      it('should render a correct symbol', ()=>{
        const mountedLineTransitType = lineTransitType()
        const iElement = mountedLineTransitType.find("i.fa")
        expect(iElement.hasClass("fa-subway")).toEqual(true)
      })
      
    })
    describe("should render ubahn correctly", ()=>{
      beforeEach(()=>{
        props.departureLine = "68"
        props.departureTransitType = "UBahn"
      })
      it('should render the correct line name', ()=>{
        const mountedLineTransitType = lineTransitType()
        const span = mountedLineTransitType.find('.departureLine')
        // console.log('SPAN: ' + span.text())
        expect(span.text()).toEqual('68')
      })
      it('should render a correct symbol', ()=>{
        const mountedLineTransitType = lineTransitType()
        const iElement = mountedLineTransitType.find("i.fa")
        expect(iElement.hasClass("fa-train")).toEqual(true)
      })
      
    })
  })
  // SNAPSHOT TESTS
  
});



//TEST DATA
// <LineTransitType
//             departureLine={this.props.departureLine}
//             departureTransitType={this.props.departureTransitType}
//           />
