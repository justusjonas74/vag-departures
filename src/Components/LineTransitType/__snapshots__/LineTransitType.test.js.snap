// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LineTransitType should render ubahn correctly renders correctly 1`] = `
<span
  className="departureTransitType"
>
  <i
    aria-hidden="true"
    className="fa fa-train"
  />
  <span
    className="departureLine ubahn"
  >
    68
  </span>
</span>
`;

exports[`LineTransitType when props are defined should render bus correctly renders correctly 1`] = `
<span
  className="departureTransitType"
>
  <i
    aria-hidden="true"
    className="fa fa-bus"
  />
  <span
    className="departureLine bus"
  >
    68
  </span>
</span>
`;

exports[`LineTransitType when props are defined should render tram correctly renders correctly 1`] = `
<span
  className="departureTransitType"
>
  <i
    aria-hidden="true"
    className="fa fa-subway"
  />
  <span
    className="departureLine tram"
  >
    68
  </span>
</span>
`;
